README.md

```diff
@@ -11,25 +12,155 @@ This operator is in charge to configure the idp client service.
-and then delete the example with `kubectl delete -f test/config/example/github-authrealm.yaml`
+popd
+```  
 
-This example creates a placement with a matchlabel `authdeployment: east` and so you have to add that label in the managedcluster you want to be managed by the authrealm AND add a label `cluster.open-cluster-management.io/clusterset: cluster-sample` in it.
+After running these commands, various pods on the OCP hub will restart in order to use the new certificate.  Wait a few minutes for all the pods to restart.  The OCP UI Overview page can be used to check the overall health of OCP.
 
-```yaml
-    authdeployment: east
-    cluster.open-cluster-management.io/clusterset: clusterset-sample
+NOTE: To use the ACME certificate process, you must have Amazon AWS credentials to allow a Route53 domain to
+be added for certificate verification during creation.
+
+
+## Setup GitHub OAuth
+Use GitHub as the OAuth provider.
+1. On github.com, go to Settings > Developer Settings > OAuth Apps
+(The shortcut is https://github.com/settings/developers)
+
+2. Add a `New OAuth App`.  Copy the `Client ID` and `Client Secret` values.  You will need them later on.
+
+3. Fill in the `Application name` with something to help identify the owner and hub it will be used for.
+NOTE: If you have more than one hub, each one will need it's own entry.
+
+4. Fill in `Homepage URL` and `Authorization callback URL` with the hub console URL.  
+(NOTE: A little bit later we will correct the `Authorization callback URL` value once we have the generated value.)
+
+5. Click `Register Application`
+
+NOTE: You will need to return to the GitHub OAuth a little bit later to correct the `Authorization callback URL`, once the value is generated for you.
+
+
+## Install the operator
+1. Clone this repo
+```bash
+git clone https://github.com/identitatem/idp-mgmt-operator.git
+cd idp-mgmt-operator
+```
+2. Login to your Red Hat Advanced Cluster Management or Multi Cluster Engine hub
+3. Verify the hub cluster
+```bash
+oc cluster-info
+```
+3. From the cloned idp-mgmt-operator directory:
+```bash
+make deploy
 ```
 
-# Undeploy the operator
+## Create an AuthRealm, ManagedClusterSet, etc
+We need to create a few custom resources to get idp-mgmt-operator to trigger.  Use the `generate-cr.sh` script in
+the `hack` directory to generate a yaml file with all the CRs needed.  
 
-`make undeploy`
\ No newline at end of file
+1. First setup the required environment variables:
+```bash
+GITHUB_APP_CLIENT_ID={your GitHub App Client ID}
+GITHUB_APP_CLIENT_SECRET={your GitHub App Client Secret}
+IDP_NAME=sample-idp
+NAME=authrealm-sample
+NS=authrealm-sample-ns
+```
+
+2. run the script
+```bash
+cd hack
+./generate-cr.sh
+```  
+
+The `generates-cr.sh` script will create the yaml and output the name of the yaml file.  In addition, it will
+provide values that need to be copied back into the GitHub Auth App.  
```

Comment:
Should we not put that yaml in a temp directory or prefix it by something we can put in the .gitignore

Reply:
Changed to /tmp